src/BayesianOptimization.jl

```diff
@@ -164,5 +232,80 @@ function _evaluate_function(o, x)
     end
     return y
 end
+"""
+    optimize(f, lowerbounds, upperbounds; <keyword arguments>)
+
+Find an optimizer between `lowerbounds` and `upperbounds` of a function `f` using default parameters.
+
+Default parameters can be overwritten by passing respective values as keyword arguments.
+
+# Arguments
+
+TODO: document defaults, keywords: 
+model, acquisition, modeloptimizer, sense, maxiterations, maxduration, acquisitionoptions, repetitions, verbosity, initializer_iterations, initializer
+"""
+function optimize(f, lowerbounds, upperbounds; optkwargs...)
+    args, kwargs = merge_with_defaults(f, lowerbounds, upperbounds, optkwargs)
+    opt = BOpt(args...; kwargs...)
+    boptimize!(opt)
+end
+"""
+    merge_with_defaults(f, lowerbounds, upperbounds, optkwargs)
+"""
+function merge_with_defaults(f, lowerbounds, upperbounds, optkwargs)
+    # the same order of args as in BOpt constructor
+    args_keys = [:model, :acquisition, :modeloptimizer]
+    kwargs_keys = [
+        :sense,
+        :maxiterations,
+        :maxduration,
+        :acquisitionoptions,
+        :repetitions,
+        :verbosity,
+        :initializer_iterations,
+        :initializer,
+    ]
+    # check if optkwargs contains only valid keyword arguments
+    issubset(keys(optkwargs), union(args_keys, kwargs_keys)) ||
+        throw(ArgumentError("use of unsupported keyword arguments"))
+
+    length(lowerbounds) == length(upperbounds) ||
+        throw(ArgumentError("length of lowerbounds does not match length of upperbounds"))
+    inputdimension = length(lowerbounds)
+
+    # TODO: come up with reasonable optdefaults
+    # optdefaults can be extended to eventually overwrite default kwargs when passed to BOpt constructor
+    optdefaults = (
```

Comment:
It could make sense to move the default options to a separate function `default_options(inputdimension)` and replace the `@assert`s with tests. But I don't have a strong opinion on this.

Reply:
Okay cool, so I will keep it like it is for now.